homeassistant/components/mpd/media_player.py

```diff
@@ -265,16 +268,44 @@ def media_album_name(self):
     @property
     def media_image_hash(self):
         """Hash value for media image."""
-        if file := self._currentsong.get("file"):
-            return hashlib.sha256(file.encode("utf-8")).hexdigest()[:16]
-
-        return None
+        return self._media_image_hash
 
     async def async_get_media_image(self):
         """Fetch media image of current playing track."""
         if not (file := self._currentsong.get("file")):
             return None, None
+        response = await self._async_get_file_image_response(file)
+        if not response:
+            return None, None
+
+        image = bytes(response.get("binary"))
```

Comment:
`bytes` will error if it's passed `None`. If we know that `"binary"` is present in the `response` dict we should use `dict[key]` to access it.
```suggestion
        image = bytes(response["binary"])
```